scripts/west_commands/completion/west-completion.zsh

```diff
@@ -0,0 +1,353 @@
+#compdef west
+# Ensure this works also when being source-ed
+compdef _west west
+
+typeset -A -g _opt_args
+
+_west_cmds() {
+  local -a builtin_cmds=(
+	'init[create a west workspace]'
+	'update[update projects described in west manifest]'
+	'list[print information about projects]'
+	'manifest[manage the west manifest]'
+	'diff["git diff" for one or more projects]'
+	'status["git status" for one or more projects]'
+	'forall[run a command in one or more local projects]'
+	'config[get or set config file values]'
+	'topdir[print the top level directory of the workspace]'
+	'help[get help for west or a command]'
+  )
+
+  local -a zephyr_ext_cmds=(
+	'completion[display shell completion scripts]'
+	'boards[display information about supported boards]'
+	'build[compile a Zephyr application]'
+	'sign[sign a Zephyr binary for bootloader chain-loading]'
+	'flash[flash and run a binary on a board]'
+	'debug[flash and interactively debug a Zephyr application]'
+	'debugserver[connect to board and launch a debug server]'
+	'attach[interactively debug a board]'
+	'zephyr-export[export Zephyr installation as a CMake config package]'
+	'spdx[create SPDX bill of materials]'
+	'blobs[work with binary blobs]'
+  )
+
+  local -a all_cmds=(${builtin_cmds} ${zephyr_ext_cmds})
+#west topdir &>/dev/null
+#if [ $? -eq 0 ]; then
+  _values "west command" $all_cmds
+#else
+#  _values "west command" $builtin_cmds
+#fi
```

Comment:
do we need such comments? same applies to other places

Reply:
Fixed, replaced with env var.